### ISSUE TITLE
Fixed Crash | Check Pull Requests

### ISSUE THREAD

**Issue by XyloFD2** [minijaham/ObsidianBreakerTNT#2]
[12:08:25.471] [Server thread/CRITICAL]: ErrorException: "Undefined property: minijaham\ObsidianBreakerTNT\ObsidianData::$block" (EXCEPTION) in "plugins/ObsidianBreakerTNT.phar/src/ObsidianData" at line 31

**Comment by XyloFD2** [comment]
I fixed it, I made a pull request with both of the fixes for the main.php and the ObsidianData.php file.

**Comment by XyloFD2** [comment]
Also theres a issue/suggestion 
![image](https://github.com/minijaham/ObsidianBreakerTNT/assets/104400758/959816a5-66da-434b-867c-7651ddb80f75)

**Comment by XyloFD2** [comment]
The tnt will only break the Obsidian if it is to land onto a block right next to the obsidian if the obsdiain blocks that are on the side of the tall wall werent there and the tnt was to explode on the wall it wouldnt damage it.

**Comment by minijaham** [comment]
> Reference in

Weird. It should be blowing every block that the TNT touches.

**Comment by minijaham** [comment]
> [12:08:25.471] [Server thread/CRITICAL]: ErrorException: "Undefined property: minijaham\ObsidianBreakerTNT\ObsidianData::$block" (EXCEPTION) in "plugins/ObsidianBreakerTNT.phar/src/ObsidianData" at line 31

This error should be fixed in the latest commit. Sorry about that haha

**Comment by XyloFD2** [comment]
> > Reference in
> 
> Weird. It should be blowing every block that the TNT touches.

Yea Its weird, Cause the tnt has to land a block for it to actually do dmg to the Obsdiain, Could you try fixing it, Cause I don't know what that issue is

**Comment by XyloFD2** [comment]
Also, I got a question, Is there a way to make it break more than 1 block? after the tnt explodes?

**Comment by XyloFD2** [comment]
like make the TNT blast radius bigger or something for the tnt breaking the obsidian

**Comment by minijaham** [comment]
> Also, I got a question, Is there a way to make it break more than 1 block? after the tnt explodes?

Yeah. You just need to use a loop from the block that the TNT touches as well.

**Comment by minijaham** [comment]
> > Also, I got a question, Is there a way to make it break more than 1 block? after the tnt explodes?
> 
> Yeah. You just need to use a loop from the block that the TNT touches as well.

Or actually have a detection for the TNT explosion, but not sure how that can be implemented. I need to look.

**Comment by XyloFD2** [comment]
> > > Also, I got a question, Is there a way to make it break more than 1 block? after the tnt explodes?
> > 
> > 
> > Yeah. You just need to use a loop from the block that the TNT touches as well.
> 
> Or actually have a detection for the TNT explosion, but not sure how that can be implemented. I need to look.

If there is a way Could you possibly implement it into the plugin? as a config option?

**Comment by minijaham** [comment]
> > > > Also, I got a question, Is there a way to make it break more than 1 block? after the tnt explodes?
> 
> > > 
> 
> > > 
> 
> > > Yeah. You just need to use a loop from the block that the TNT touches as well.
> 
> > 
> 
> > Or actually have a detection for the TNT explosion, but not sure how that can be implemented. I need to look.
> 
> 
> 
> If there is a way Could you possibly implement it into the plugin? as a config option?

I can, but I can't test it atm 😭

**Comment by XyloFD2** [comment]
> > > > > Also, I got a question, Is there a way to make it break more than 1 block? after the tnt explodes?
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > Yeah. You just need to use a loop from the block that the TNT touches as well.
> > 
> > 
> > > 
> > 
> > 
> > > Or actually have a detection for the TNT explosion, but not sure how that can be implemented. I need to look.
> > 
> > 
> > If there is a way Could you possibly implement it into the plugin? as a config option?
> 
> I can, but I can't test it atm 😭

I can test for you, if you have discord we can communicate through there aswell

**Comment by XyloFD2** [comment]
My discord is xylofd if you wanna add me
